lib/msal-common/src/client/BaseClient.ts

```diff
@@ -79,85 +75,60 @@ export abstract class BaseClient {
+            : this.defaultAuthorityInstance;
 
-    /**
-     * Given an authorization code, it will perform a token exchange using cached values from a previous call to
-     * createLoginUrl() or createAcquireTokenUrl(). You must call this AFTER using one of those APIs first. You should
-     * also use the handleFragmentResponse() API to pass the codeResponse to this function afterwards.
-     * @param codeResponse
-     */
-    abstract async acquireToken(codeResponse: CodeResponse): Promise<TokenResponse>;
+        try {
+            await authority.resolveEndpointsAsync();
+        } catch (error) {
+            throw ClientAuthError.createEndpointDiscoveryIncompleteError(error);
+        }
 
-    /**
-     * Retrieves a token from cache if it is still valid, or uses the cached refresh token to renew
-     * the given token and returns the renewed token. Will throw an error if login is not completed (unless
-     * id tokens are not being renewed).
-     * @param request
-     */
-    abstract async renewToken(request: TokenRenewParameters): Promise<TokenResponse>;
+        return authority;
+    }
 
     /**
-     * Use to log out the current user, and redirect the user to the postLogoutRedirectUri.
-     * Default behaviour is to redirect the user to `window.location.href`.
-     * @param authorityUri
+     * Creates default headers for requests to token endpoint
      */
-    abstract async logout(authorityUri?: string): Promise<string>;
-
-    // #endregion
-
-    // #region Getters and Setters
+    protected createDefaultTokenRequestHeaders(): Map<string, string> {
+        const headers = new Map<string, string>();
+        RequestUtils.addContentTypeHeader(headers);
+        RequestUtils.addLibrarydataHeaders(headers);
+        return headers;
+    }
 
     /**
-     * Returns the signed in account
-     * (the account object is created at the time of successful login)
-     * or null when no state is found
-     * @returns {@link Account} - the account object stored in MSAL
+     * Http post to token endpoint
+     * @param tokenEndpoint
+     * @param queryString
+     * @param headers
      */
-    getAccount(): Account {
-        if (this.account) {
-            return this.account;
-        }
-
-        // Get id token and client info from cache
-        const rawIdToken = this.cacheStorage.getItem(PersistentCacheKeys.ID_TOKEN);
-        const rawClientInfo = this.cacheStorage.getItem(PersistentCacheKeys.CLIENT_INFO);
-
-        if(!StringUtils.isEmpty(rawIdToken) && !StringUtils.isEmpty(rawClientInfo)) {
-            const idToken = new IdToken(rawIdToken, this.cryptoObj);
-            const clientInfo = buildClientInfo(rawClientInfo, this.cryptoObj);
-
-            this.account = Account.createAccount(idToken, clientInfo, this.cryptoObj);
-            return this.account;
-        }
-
-        // if login is not yet done, return null
-        return null;
+    protected async executePostToTokenEndpoint(
```

Comment:
can we add return type here to make it clear we return a promise?

Reply:
Return type is actually a couple of lines down.